### PR TITLE
Updated the resource settings models of following tracked NRP resources

### PR DESCRIPTION
**Pull request by innosam**
1. Updated the resource settings models of following tracked NRP resources:
- Network Interfaces
- Network Security Groups
- Virtual Networks
- Loadbalancers
- Public IP

2. Fix GO SDK generation.

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/regionmove/resource-manager/Microsoft.Migrate/preview/2019-10-01-preview/regionmovecollection.json [hunk 1]
```diff
@@ -1654,6 +1654,20 @@
       },
       "x-ms-discriminator-value": "Microsoft.Compute/availabilitySets"
     },
+    "SubnetsResourceSettings": {
```
Comment (ArcturusZhang) [inline specification/regionmove/resource-manager/Microsoft.Migrate/preview/2019-10-01-preview/regionmovecollection.json:1657]: For semantic considerations, could we rename this to `SubnetResourceSetting` if possible?

File: specification/regionmove/resource-manager/Microsoft.Migrate/preview/2019-10-01-preview/regionmovecollection.json [hunk 2]
```diff
@@ -1662,19 +1676,63 @@
...
         }
       ],
-      "properties": {},
+      "properties": {
+        "enableDdosProtection": {
+          "description": "Gets or sets a value indicating whether gets or sets whether the\r\nDDOS protection should be switched on.",
+          "type": "boolean",
+          "x-nullable": true
+        },
+        "addressSpace": {
```
Comment (ArcturusZhang) [inline specification/regionmove/resource-manager/Microsoft.Migrate/preview/2019-10-01-preview/regionmovecollection.json:1685]: For semantic considerations, maybe this makes more sense if it is plural? For instance `addressSpaces`

**Review by ArcturusZhang — COMMENTED** [review]
LGTM with some minor comments

File: specification/regionmove/resource-manager/Microsoft.Migrate/preview/2019-10-01-preview/regionmovecollection.json (hunk 2 above)
Comment (innosam) [reply specification/regionmove/resource-manager/Microsoft.Migrate/preview/2019-10-01-preview/regionmovecollection.json:1685]: We have aligned here with the network model as per the Microsoft.Network RP.

Refer: https://docs.microsoft.com/en-us/rest/api/virtualnetwork/virtualnetworks/get#addressspace

File: specification/regionmove/resource-manager/Microsoft.Migrate/preview/2019-10-01-preview/regionmovecollection.json (hunk 1 above)
Comment (innosam) [reply specification/regionmove/resource-manager/Microsoft.Migrate/preview/2019-10-01-preview/regionmovecollection.json:1657]: taken.

**Comment by ArcturusZhang** [comment]
As identified in the automated checks there are breaking changes, Please follow Azure Breaking changes approval request form as defined [Breaking Change Process](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37673/Retirements-and-Breaking-Changes)

**Comment by ArcturusZhang** [comment]
approved breaking changes since this is a preview api within one year

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**